### DISCUSSION TITLE
Best way to play Counter-Strike 2 stretched?

### DISCUSSION THREAD

**Discussion by CNR0706** [hyprwm/Hyprland#5685] [General]
When running the game in native Wayland (SDL_VIDEO_DRIVER=wayland) it works almost perfectly. Only issue is that Tearing does not work like this (even though there is nothing else visible on the monitor).

When using the csgo-vulkan-fix plugin Tearing works but everything looks super pixelated for some reason. Especially text.

Has anyone found a better solution?

gamescope and rootful XWayland don't work well unfortunately.

**Comment by vaxerski** [comment]
I play stretched with csgo-vk-fix and everything works smoothly. Text is pixelated because it's stretched...?

**Reply by CNR0706** [reply]
It looks _a lot_ better when not using the plugin so I guess this is due to different scaling algorithms or something? I don't really know.

Do you know why fullscreening CS:2 when running under Wayland at a non-native resolution doesn't cause Hyprland to start Tearing? Resolving this would fix everything. I tried some very basic stuff already but I wasn't successful in resolving this.

**Reply by vaxerski** [reply]
no clue, would have to debug. Only thing that comes to mind is cs2 having more than one surface, but idk why it'd do that, so I doubt its the problem

**Reply by CNR0706** [reply]
The second surface should show up in hyprctl clients right? If that's the case: CS:2 does not seem to create a second surface.

Should I open an issue about this? CS:2 doesn't seem to be the only application that's a bit wonky when it comes to Tearing.

**Reply by vaxerski** [reply]
No, it won't show in hyprctl clients.

**Reply by 3ri4nG0ld** [reply]
What @CNR0706  says about it looking pixelated happens, in my case it is very evident in the crosshairs, which look displaced.

![Captura de pantalla_20240531_105759](https://github.com/hyprwm/Hyprland/assets/24411137/fc702a19-7ba2-4f99-83f2-6ae53942e931)

PD: I use 1440x1080 resolution
In X11 or using gamescope it looks perfect, it is only displaced with the plugin.

**Reply by 3ri4nG0ld** [reply]
This is how it looks like from X11
![Captura de pantalla_2024-05-31_23-10-26 (1)](https://github.com/hyprwm/Hyprland/assets/24411137/bd84e437-1129-431b-8bad-80a210bb20bb)

In both cases the game is in 1440x1080 (4:3).
but you can see that using the plugin it is not the same.

**Reply by kito30** [reply]
I also experience blurry and pixelated in cs2

**Comment by jindongjie** [comment]
Did steam support wayland？my steam work bad in xwayland：（

**Reply by CNR0706** [reply]
No, but most GoldSrc, Source and Source 2 games do.

**Comment by zeptic99** [comment]
@vaxerski How do you play cs2 stretched? For me, using csgo-vulkan-fix has black bars when i go fullscreen. My solution till now had been gamescope but it adds noticeable input lag.

**Reply by vaxerski** [reply]
csgo-vulkan-fix.

**Reply by zeptic99** [reply]
Yes, I am using the plugin. The game still has black bars (on 1080x1080 res)

**Reply by zeptic99** [reply]
I apologize for my incompetency, I had to change "cs2" to "SDL Application" in the config, and now it works.

**Reply by vaxerski** [reply]
yeah, some recent cs2 update broke that one

**Reply by Locotir** [reply]
> I apologize for my incompetency, I had to change "cs2" to "SDL Application" in the config, and now it works.

Where exactly you put the config? On the normal hyprland.conf?

**Comment by IceDBorn** [comment]
This is my config
` plugin { csgo-vulkan-fix { res_w = 1440 res_h = 1080 class = SDL Application fix_mouse = true } }
`
and these are my launch options `mangohud gamemoderun %command% -vulkan -nojoy -novid -threads 12 -window -w 1440 -h 1080`. I've also set the game to `wayland` instead of `x11`. It is stretched, but my mouse does not behave as it should.

My monitor is 16:9, 1080p.

**Reply by IceDBorn** [reply]
> how set the game to wayland

Follow the path to `Counter-Strike Global Offensive/game`, edit `cs2.sh` and search for `x11`. The line you're looking for is `export SDL_VIDEO_DRIVER=x11`, to set the game to wayland just change the line like this `export SDL_VIDEO_DRIVER=wayland`.

**Reply by DoYouSeeAmin** [reply]
I found a fix for this problem, I did everything like you did and I have the same problem with the mouse, to solve it I need to set windowed in the game settings and return to fullscreen and that’s it "now its not working"

**Reply by DoYouSeeAmin** [reply]
after the latest update cs 2 gamescope works great try this. my settings `gamemoderun  gamescope -e -W 1280 -H 960 -r 180  --force-grab-cursor -S stretch -- %command%`

**Reply by shibahex** [reply]
> > how set the game to wayland
> 
> Follow the path to `Counter-Strike Global Offensive/game`, edit `cs2.sh` and search for `x11`. The line you're looking for is `export SDL_VIDEO_DRIVER=x11`, to set the game to wayland just change the line like this `export SDL_VIDEO_DRIVER=wayland`.

This ^^^ is the fix that took me like a week to find, ty.

**Comment by kito30** [comment]
When I play cs2 with 4:3 the mouse movement feel weird, when you look up the mouse always stuck at somepoint

**Reply by amadejkastelic** [reply]
Are you using the csgo-vulkan-fix plugin? Did you enable the `fix_mouse` option?

**Reply by kito30** [reply]
I did have vulkan enable but how can I enable fix_mouse option

**Reply by amadejkastelic** [reply]
https://github.com/hyprwm/hyprland-plugins/tree/main/csgo-vulkan-fix
Also make sure the game launches in windowed mode and fullscreen it manually after reaching the main menu.

**Reply by kito30** [reply]
It worked, thank you

**Comment by kito30** [comment]
Im also experience low fps on CS2. Usually around 90 110fps while on windows is 150 190fps

**Reply by DoYouSeeAmin** [reply]
try switch csgo to wayland but in wayland be bug when ur mouse does not behave as it should in 4/3. u can fix that by set in settings of game windowed and return to fullscreen

**Reply by kito30** [reply]
Already try to set to windowed and go back to fullscreen. The mouse still not hiding.

**Reply by DoYouSeeAmin** [reply]
yes i now checked this. today update broke wayland in csgo now its not working

**Reply by kito30** [reply]
Men guess Deadlock now

**Reply by DoYouSeeAmin** [reply]
btw u can have less fps than on windows because u use wayland and cs2 run in xwayland

**Comment by DoYouSeeAmin** [comment]
after the latest update cs 2 gamescope works great try this. my settings `gamemoderun  gamescope -e -W 1280 -H 960 -r 180  --force-grab-cursor -S stretch -- %command%`

**Comment by Locotir** [comment]
My cursor is not grabbing, I also put this config on hyprland.conf:

plugin {
  csgo-vulkan-fix {
    res_w = 1280
    res_h = 960
    class = "SDL Application"
    fix_mouse = true
  }
}

Is the only way to fix it gamescope @vaxerski ?

**Comment by Locotir** [comment]
The problem is setting the SDL to wayland, then the cursor always still visible

**Comment by Locotir** [comment]
I finally did it, not using wayland & enabling at least antialiasing x4 on the game, now if I fullscreen cs2, is streched

**Comment by HateJoysticks** [comment]
Hey guys, I fixed It by disabling Anti-aliasing, or setting it off to CMAA
no need for gamescope!

**Reply by CNR0706** [reply]
The fact that this actually works is just bizarre. Thanks though!

**Reply by patrickhaahr** [reply]
this works
